### PR TITLE
Get tx

### PR DESCRIPTION
**Pull request by noamnelke**

### CODE REVIEW

**Review by IlyaVi — APPROVED** [review]

File: api/grpc_server.go [hunk 1]
```diff
@@ -21,15 +22,74 @@ import (
...
+	TxMempool     *miner.TxMempool // TX Mempool
+	Mining        MiningAPI        // ATX Builder
+	Oracle        OracleAPI
+	GenTime       GenesisTimeAPI
+	LayerDuration time.Duration
+	Logging       LoggingAPI
+}
+
+func (s SpacemeshGrpcService) getTransaction(txId types.TransactionId) (*types.Transaction, *types.LayerID, pb.TxStatus, error) {
+	tx, err := s.Tx.GetTransaction(txId) // have we seen this transaction in a block?
```
Comment (antonlerner) [inline api/grpc_server.go:39]: IMO, this logic seems to belong inside as part of the MVC pattern. the grpc is merley the view builder

File: api/grpc_server.go [hunk 2]
```diff
@@ -21,15 +22,74 @@ import (
...
+	Tx            TxAPI            // Mesh
+	TxMempool     *miner.TxMempool // TX Mempool
+	Mining        MiningAPI        // ATX Builder
+	Oracle        OracleAPI
+	GenTime       GenesisTimeAPI
+	LayerDuration time.Duration
+	Logging       LoggingAPI
+}
+
+func (s SpacemeshGrpcService) getTransaction(txId types.TransactionId) (*types.Transaction, *types.LayerID, pb.TxStatus, error) {
```
Comment (antonlerner) [inline api/grpc_server.go]: I'd revise to getTransactionStatue since this function return more than a simple tx

File: api/grpc_server.go (hunk 1 above)
Comment (noamnelke) [reply api/grpc_server.go:39]: Hmm... I see what you mean, but there's no place that currently has access to all the databases required. The mesh has no knowledge of the mempool and vice-versa.

I could create a transactions controller or something, but it feels like overkill for this little logic.

WDYT?

File: api/grpc_server.go (hunk 2 above)
Comment (noamnelke) [reply api/grpc_server.go]: done

File: api/grpc_server.go [hunk 3]
```diff
@@ -21,15 +22,74 @@ import (
...
+	Tx            TxAPI            // Mesh
+	TxMempool     *miner.TxMempool // TX Mempool
+	Mining        MiningAPI        // ATX Builder
+	Oracle        OracleAPI
+	GenTime       GenesisTimeAPI
+	LayerDuration time.Duration
+	Logging       LoggingAPI
+}
+
+func (s SpacemeshGrpcService) getTransactionAndStatus(txId types.TransactionId) (*types.Transaction, *types.LayerID, pb.TxStatus, error) {
```
Comment (antonlerner) [inline api/grpc_server.go:38]: please add unit tests

**Pull request merged**